### DISCUSSION TITLE
Native Postgres Handler in databend

### DISCUSSION THREAD

**Discussion by sundy-li** [databendlabs/databend#10742] [General]
Postgresql is a very popular database, Vote `+1/-1` If you like to have pg protocol support in databend.

https://github.com/sunng87/pgwire could be a good crate option.

**Comment by flaneur2020** [comment]
I'd sugguest write a standalone proxy middleware in another repository over databend's flight API:

- the flight protocol is easier to integrate with cloud native infrastructure like services in Kubernetes, easy to load balance and offering services to users. (on the oppose side, it's hard to expose an Postgres port to customers in kubernetes
- the pg-over-flight middleware can be used in other projects, and we can get contributions from whom use it.
- we can also deploy the middleware as a sidecar aside with databend, which offers the benefits of the postgres protocol, but simplifies the maintence work about databend's protocols (there have already too many protocols, makes it harder for users to choose the best maintained one).

**Reply by sundy-li** [reply]
Agree， we should maintain a main protocol.

**Reply by sundy-li** [reply]
But it needs another service to deploy.

**Reply by flaneur2020** [reply]
IMHO, deploying a sidecar is cheap in kuberenetes.

In the virtual machine deployments, a gateway like PG proxy is also not hard to deploy at all. It's totally stateless, can be started up with a plain backend list config.
